src/SFA.DAS.AssessorService.Application.Api.External/Infrastructure/ApiClient.cs

```diff
@@ -15,23 +22,81 @@ public class ApiClient
         private readonly ILogger<ApiClient> _logger;
         private readonly ITokenService _tokenService;
 
+        private readonly JsonSerializerSettings _jsonSettings = new JsonSerializerSettings
+        {
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+            NullValueHandling = NullValueHandling.Ignore
+        };
+
         public ApiClient(HttpClient client, ILogger<ApiClient> logger, ITokenService tokenService)
         {
             _client = client;
             _logger = logger;
             _tokenService = tokenService;
         }
 
-        private async Task<T> Get<T>(string uri)
+        protected async Task<T> Get<T>(string uri)
         {
             _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _tokenService.GetToken());
             var res = await _client.GetAsync(new Uri(uri, UriKind.Relative));
             return await res.Content.ReadAsAsync<T>();
         }
 
-        public async Task<List<OrganisationResponse>> GetAllOrganisations()
+        protected async Task<U> PostPutRequestWithResponse<T, U>(HttpRequestMessage requestMessage, T model)
+        {
+            var serializeObject = JsonConvert.SerializeObject(model);
+            requestMessage.Content = new StringContent(serializeObject,
+                System.Text.Encoding.UTF8, "application/json");
+
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _tokenService.GetToken());
+
+            using (var response = await _client.SendAsync(requestMessage))
+            {
+                var json = await response.Content.ReadAsStringAsync();
+
+                if (response.StatusCode == HttpStatusCode.OK
+                    || response.StatusCode == HttpStatusCode.Created
+                    || response.StatusCode == HttpStatusCode.NoContent)
+                {
+                    return await Task.Factory.StartNew<U>(() => JsonConvert.DeserializeObject<U>(json, _jsonSettings));
+                }
+                else
+                {
+                    _logger.LogInformation($"HttpRequestException: Status Code: {response.StatusCode} Body: {json}");
+                    throw new HttpRequestException(json);
+                }
+            }
+        }
```

Comment:
This looks like it's copied from the old ApiClient project.  Have a look at how I've replaces the Put / Post in the ON-674... branch.

Reply:
I checked that branch and it looks the same and same as Web.Staff . Are you expecting me to use  SFA.DAS.AssessorService.Application.Api.Client.Clients.ApiClientBase ?